service/worker/service.go

```diff
@@ -340,26 +343,26 @@ func (s *Service) startFailoverManager() {
 	}
 }
 
-func (s *Service) ensureSystemDomainExists() {
-	_, err := s.GetMetadataManager().GetDomain(context.Background(), &persistence.GetDomainRequest{Name: common.SystemLocalDomainName})
+func (s *Service) ensureDomainExists(domain string) {
+	_, err := s.GetMetadataManager().GetDomain(context.Background(), &persistence.GetDomainRequest{Name: domain})
 	switch err.(type) {
 	case nil:
 		// noop
 	case *shared.EntityNotExistsError:
-		s.GetLogger().Info("cadence-system domain does not exist, attempting to register domain")
-		s.registerSystemDomain()
+		s.GetLogger().Info(fmt.Sprintf("domain %s does not exist, attempting to register domain", domain))
+		s.registerSystemDomain(domain)
 	default:
-		s.GetLogger().Fatal("failed to verify if cadence system domain exists", tag.Error(err))
+		s.GetLogger().Fatal("failed to verify if system domain exists", tag.Error(err))
 	}
 }
 
-func (s *Service) registerSystemDomain() {
+func (s *Service) registerSystemDomain(domain string) {
 
 	currentClusterName := s.GetClusterMetadata().GetCurrentClusterName()
 	_, err := s.GetMetadataManager().CreateDomain(context.Background(), &persistence.CreateDomainRequest{
 		Info: &persistence.DomainInfo{
-			ID:          common.SystemDomainID,
-			Name:        common.SystemLocalDomainName,
+			ID:          uuid.New().String(),
```

Comment:
probably better to stick to the previous approach and use constants for safety sake?

Reply:
agree it might be safer for sql. changed back.